topics/dev/tutorials/core-contributing/tutorial.md

```diff
@@ -0,0 +1,219 @@
+
+> ### {% icon question %} Questions about Mapping
+>
+> 1. What should the SQL Alchemy model named corresponding to the table ``user_favorite_extension`` based on other examples in the file.
+> 2. What table stores Galaxy's user records?
+> 3. What is another simple table with a relationship with the Galaxy's user table?
+>
+> > ### {% icon solution %} Solution
+> > 1. ``UserFavoriteExtension`` 
+> > 2. ``galaxy_user``
+> > 3. An example table might be the ``user_preference`` table.
+> {: .solution }
+{: .question }
+
+Implement the required changes ``mapping.py`` to add a mapping for
+the proposed ``user_favorite_extension`` table.
+
+{% include topics/dev/tutorials/core-contributing/mapping.py_diff.md %}
+
+The Python model objects used by Galaxy corresponding to these tables are
+defined in ``lib/galaxy/model/__init__.py``.
+
+Modify ``lib/galaxy/model/__init__.py`` to add a model class
+called ``UserFavoriteExtension`` as described above.
+
+{% include topics/dev/tutorials/core-contributing/__init__.py_diff.md %}
+
+There is one last database issue to consider before moving on to considering the API.
+Each successive release of Galaxy requires recipes for how to migrate old database schemes
+to updated ones. These recipes are called versions and currently implemented using SQL
+Alchemy migrate. These versions are stored in ``lib/galaxy/model/migrate``.
+
+Each of these versions is prefixed with a number 4 digit (e.g. ``0125``) to specify the linear
+order these migrations should be applied in.
+
+We've devloped a lot of abstractions around the underlying migration library we use, so it
+is probably better to find existing examples inside of Galaxy for writing migrations than
+consulting the SQL Alchemy Migrate documentation.
+
+A good example for the table we need to construct for this example is again based on the existing
+user preferences concept in Galaxy.
+
+> ### {% icon question %} Questions about Migrations
+>
+> 1. What existing Galaxy migration added the concept of user preferences to the Galaxy codebase?
+>
+> > ### {% icon solution %} Solution
+> > 1. ``lib/galaxy/model/migrate/versions/0021_user_prefs.py``
+> {: .solution }
+{: .question }
+
+
+Add a new file to ``lib/galaxy/model/migrate/versions/`` prefixed appropriately.
+
+{% include topics/dev/tutorials/core-contributing/0177_add_user_favorite_extensions.py_diff.md %}
+
+With the database model in place, we need to start adding the rest of
+the Python plubming required to implement this feature. We will do
+this with a test-driven approach and start by implementing an API test
+that exercises operations we would like to have available for favorite
+extensions.
+
+We will stick a test case for user extensions in
+``lib/galaxy_test/api/test_users.py`` which is a relatively
+straight-forward file that contains tests for other user API
+endpoints.
+
+Various user centered operations have endpoints under
```

Comment:
```suggestion
Various user-centered operations have endpoints under
```
Also, this sentence sounds confusing - is there something missing? (I may be wrong though)